apiserver/facades/client/spaces/spaces.go

```diff
@@ -369,22 +377,16 @@ func (api *API) ensureSpacesAreMutable(ctx stdcontext.Context) error {
 // An error is returned if it is the provider and not the Juju operator
 // that determines the space topology.
 func (api *API) ensureSpacesNotProviderSourced(ctx stdcontext.Context) error {
-	env, err := environs.GetEnviron(ctx, api.backing, environs.New)
+	env, err := api.providerGetter(ctx)
 	if err != nil {
-		return errors.Annotate(err, "retrieving environ")
+		return fmt.Errorf("getting networking environ for model: %w", err)
 	}
-
-	netEnv, ok := env.(environs.NetworkingEnviron)
-	if !ok {
-		return errors.NotSupportedf("provider networking")
```

Comment:
Maybe having this error back in for the code above would work.

Reply:
We don't need this anymore, because the providerfactory.ProviderRunner function already does this type check on the generic type passed in.

Reply:
This is not needed anymore.